src/sql/workbench/parts/dashboard/widgets/properties/propertiesWidget.component.ts

```diff
@@ -111,16 +138,103 @@ export class PropertiesWidgetComponent extends DashboardWidget implements IDashb
 		this._changeRef.detectChanges();
 	}
 
-	private parseProperties() {
-		const provider = this._config.provider;
-
-		let propertyArray: Array<Property>;
-
+	private parseConfig(): void {
 		// if config exists use that, otherwise use default
 		if (this._config.widget['properties-widget'] && this._config.widget['properties-widget'].properties) {
 			const config = <PropertiesConfig>this._config.widget['properties-widget'];
-			propertyArray = config.properties;
-		} else {
+			this.parseProperties(config.properties);
+		}
+		else if (this._config.widget['properties-widget'] && this._config.widget['properties-widget'].endpoints) {
+			this.parseEndpoints();
+		}
+		else {
+			this.parseProperties(null);
+		}
+		this._loading = false;
+	}
+
+	private parseEndpoints(): void {
+		if (!this.endpoints || this.endpoints.length === 0) {
+			this.getEndpoints();
+		}
+
+		// iterate over endpoint properties and display them
+		this.properties = [];
+		for (let i = 0; i < this.endpoints.length; i++) {
+			const endpoint = this.endpoints[i];
+			const assignProperty = {};
+			let propertyObject = this.getValueOrDefault<string>(this.endpoints[i], EndpointConstants.hyperlink.toString(), '--' || '--');
+
+			assignProperty['displayName'] = endpoint.serviceName;
+			assignProperty['value'] = propertyObject;
+			assignProperty['hyperlink'] = this.isHyperlink(endpoint.serviceName);
```

Comment:
Nit - but these could be cleaned up by initializing the object with them to begin with : 

const displayProperty: DisplayProperty = {
   displayName: endpoint.serviceName,
   value = propertyObject,
   hyperlink = this.isHyperlink(endpoint.serviceName)
};
this.properties.push(displayProperty);

Reply:
yep, this's much cleaner! I just copied what they had earlier.